### ISSUE TITLE
CLI len(response) < 2 - USB Disconnect?

### ISSUE THREAD

**Issue by chillyvee** [cosmos/ledger-cosmos#57]
Often on CLI TX signing with ledger I get the following error:

```Default sign-mode 'direct' not supported by Ledger, using sign-mode 'amino-json'.
Error: len(response) < 2
```

The ledger device may or may not display the TX information. (50/50 chance)

Is there a portion of this project we should look into to help improve?

Or is this related to needing to change the protocol used to establish a better connection with the USB device on ledger?  (U2F vs HID which might take a significant amount of work?)

**Comment by chillyvee** [comment]
This error occurs on OS X on both X64 and M1/ARM64

TLDR -> problem starts in zondax/hid but might be caused by:
   LedgerNanoX + Ledger-Cosmos + OSX + USB HID + CLI combination. 

We note a compile warning on junod
../gopath/pkg/mod/github.com/zondax/hid@v0.9.0/hidapi/mac/hid.c:693:34: warning: 'kIOMasterPortDefault' is deprecated: first deprecated in macOS 12.0 [-Wdeprecated-declarations]

We think that error message comes from:
https://github.com/cosmos/ledger-go/blob/master/ledger.go#L210

Which is a response problem from UnwrapResponseAPDU()
https://github.com/cosmos/ledger-go/blob/master/apduWrapper.go#L138

Adding a little more information to the error, we arrive at the following from time to time:
ledger nano S: len(response) < 2, "Invalid channel"

This is due to an APDU of all zeros being read every so often:
https://github.com/chillyvee/ledger-go/blob/debug/apduWrapper.go#L143

UnwrapResponseAPDU 0x101, [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0], 0x40

In some cases, disposing of this 0 packet will yield the proper packet (sequenceInt = 0) on the next read.  In other cases, the first packet seems to be gone forever.


Which may point to a problem in the upstream https://github.com/Zondax/hid READ()  where the C.hid_read somehow returns 64 bytes read, but returns a zero filled buffer.
https://github.com/Zondax/hid/blob/master/hid_enabled.go#L216
C.hid_read [64] => [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]

Noted the core problem at zondax/hid:
https://github.com/Zondax/hid/issues/6

**Comment by ftheirs** [comment]
Related to: https://github.com/Zondax/ledger-go/pull/23
